File: class2relational/src/main/java/org/atlanmod/model/dynamic/classModel/ClassLink.java
package org.atlanmod.model.dynamic.classModel;

import org.atlanmod.model.DynamicElement;
import org.atlanmod.model.DynamicLink;

import java.util.List;

public abstract class ClassLink extends DynamicLink {

    public ClassLink(String type, DynamicElement source, List<DynamicElement> target) {
        super(type, source, target);
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof ClassLink){
            ClassLink obj_ = (ClassLink) obj;
            return  this.getType().equals(obj_.getType())
                    && this.getSource().equals(obj_.getSource())
                    && this.getTarget().equals(obj_.getTarget());
        }
        return false;
    }
}

File: relational2class/src/main/scala/org/atlanmod/model/classmodel/ClassLink.scala
package org.atlanmod.model.classmodel

import org.atlanmod.model.DynamicLink

abstract class ClassLink(type_ : String, source: ClassElement, target: List[ClassElement])
  extends DynamicLink(type_, source, target){

    override def equals(o: Any): Boolean = {
        o match {
            case obj: ClassLink => obj.getType.equals(type_) & obj.getSource.equals(source) & obj.getTarget.equals(target)
            case _ => false
        }
    }

}
